cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -839,130 +846,114 @@ func buildScaleUpStatusNodeGroup(isScaleUpInProgress bool, scaleUpSafety NodeGro
 		condition.Status = api.ClusterAutoscalerUnhealthy
 	} else if !scaleUpSafety.SafeToScale {
 		condition.Status = api.ClusterAutoscalerBackoff
+		condition.BackoffInfo = api.BackoffInfo{
+			ErrorCode:    scaleUpSafety.BackoffStatus.ErrorInfo.ErrorCode,
+			ErrorMessage: scaleUpSafety.BackoffStatus.ErrorInfo.ErrorMessage,
+		}
 	} else {
 		condition.Status = api.ClusterAutoscalerNoActivity
 	}
+	if condition.Status == lastStatus.Status {
+		condition.LastTransitionTime = lastStatus.LastTransitionTime
+	} else {
+		condition.LastTransitionTime = condition.LastProbeTime
+	}
 	return condition
 }
 
-func buildScaleDownStatusNodeGroup(candidates []string, lastProbed time.Time) api.ClusterAutoscalerCondition {
-	condition := api.ClusterAutoscalerCondition{
-		Type:          api.ClusterAutoscalerScaleDown,
-		Message:       fmt.Sprintf("candidates=%d", len(candidates)),
+func buildScaleDownStatusNodeGroup(candidates []string, lastProbed time.Time, lastStatus api.ScaleDownCondition) api.ScaleDownCondition {
+	condition := api.ScaleDownCondition{
+		Candidates:    len(candidates),
 		LastProbeTime: metav1.Time{Time: lastProbed},
 	}
 	if len(candidates) > 0 {
 		condition.Status = api.ClusterAutoscalerCandidatesPresent
 	} else {
 		condition.Status = api.ClusterAutoscalerNoCandidates
 	}
+	if condition.Status == lastStatus.Status {
+		condition.LastTransitionTime = lastStatus.LastTransitionTime
+	} else {
+		condition.LastTransitionTime = condition.LastProbeTime
+	}
 	return condition
 }
 
-func buildHealthStatusClusterwide(isReady bool, readiness Readiness) api.ClusterAutoscalerCondition {
-	condition := api.ClusterAutoscalerCondition{
-		Type: api.ClusterAutoscalerHealth,
-		Message: fmt.Sprintf("ready=%d unready=%d (resourceUnready=%d) notStarted=%d longNotStarted=0 registered=%d longUnregistered=%d",
-			len(readiness.Ready),
-			len(readiness.Unready),
-			len(readiness.ResourceUnready),
-			len(readiness.NotStarted),
-			len(readiness.Registered),
-			len(readiness.LongUnregistered),
-		),
+func buildHealthStatusClusterwide(isHealthy bool, readiness Readiness, lastStatus api.ClusterHealthCondition) api.ClusterHealthCondition {
+	condition := api.ClusterHealthCondition{
+		NodeCounts: api.NodeCount{
```

Comment:
nit: This happens at least twice, maybe it would be worth having a function `NodeCountsFromReadiness(readiness Readiness) api.NodeCount`? It's less boilerplate and more importantly it makes it easier to do changes - like the one to representing unregistered that I mentioned above.

Reply:
Done.